rover_video/src/screenshot_server.hpp

```diff
@@ -0,0 +1,681 @@
+#include "rclcpp/rclcpp.hpp"
+#include "rover_msgs/msg/gps_position.hpp"
+#include "rover_msgs/srv/camera_control.hpp"
+#include "rovus_lib/macros.h"
+
+#include "opencv2/core.hpp"
+#include "opencv2/highgui.hpp"
+#include "opencv2/imgcodecs.hpp"
+
+#include <atomic>
+#include <chrono>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <thread>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include <cstdlib>
+#include <sys/stat.h>
+
+#define SCREENSHOT 1
+#define VIDEO 2
+
+constexpr uint8_t RECORDING_INTERVAL = 20;  // in seconds
+
+/* This folder is used for the functions' declarations */
+
+/*
+std::string getTempdir()
+{
+    const char* temp = std::getenv("TMPDIR");
+    if (!temp)
+        temp = std::getenv("TMP");
+    if (!temp)
+        temp = std::getenv("TEMP");
+    if (!temp)
+        temp = "/tmp";  // Default for Ubuntu
+    return std::string(temp);
+}
+*/
+
+class Recording
+{
+  public:
+    bool startRecording();
+    bool recordFrame();
+
+    std::string getURL()
+    {
+        return this->camURL;
+    }
+
+    cv::Mat getFrame()
+    {
+        return this->frame;
+    }
+
+    uint8_t getFPS()
+    {
+        return this->fps;
+    }
```

Comment:
CG: Functions with no arguments should take void as argument.
CG: Use this-> when calling method inside a class. (for private member it's not necesssary since they should use a prefix "_" in their name)


Also they could all be defined const since they don't modify any value in the class: 

```suggestion
    bool startRecording(void);
    bool recordFrame(void);

    std::string getURL(void) const
    {
        return camURL;
    }

    cv::Mat getFrame(void) const
    {
        return frame;
    }

    uint8_t getFPS(void) const
    {
        return fps;
    }
```